File: book/05_struct_data/src/triangle.go
package main

import "fmt"

type Rectangle struct {
	height int
	width  int
}

func newRectangle(height, width int) *Rectangle {
	return &Rectangle{height, width}
}

func (r *Rectangle) area() int {
	return r.height * r.width
}

func main() {
	rect := newRectangle(50, 60)
	fmt.Printf("The area of the rectangle is %d square pixels.", rect.area())
}

File: book/05_struct_data/src/triangle.rs
struct Rectangle {
    height: usize,
    width: usize,
}

impl Rectangle {
    // Static Methods, yey
    fn new(height: usize, width: usize) -> Self {
        Self { height, width }
    }

    fn area(&self) -> usize {
        self.width * self.height
    }
}

fn main() {
    let rect = &Rectangle::new(50, 60);

    println!(
        "The area of the rectangle is {} square pixels.",
        rect.area()
    );
}
